oidc/examples/cli/main.go

```diff
@@ -71,22 +74,86 @@ func main() {
 	usePKCE := flag.Bool("pkce", false, "use the implicit flow")
 	maxAge := flag.Int("max-age", -1, "max age of user authentication")
 	scopes := flag.String("scopes", "", "comma separated list of additional scopes to requests")
+	useTestProvider := flag.Bool("use-test-provider", false, "use the test oidc provider")
 
 	flag.Parse()
 	if *useImplicit && *usePKCE {
 		fmt.Fprint(os.Stderr, "you can't request both: -implicit and -pkce")
 		return
 	}
 
+	if (*useImplicit || *implicitAccessToken || *scopes != "") && *useTestProvider {
+		fmt.Fprint(os.Stderr, "you can't use the implicit flow, PKCE or scopes with the test provider")
+		return
+	}
+
 	optScopes := strings.Split(*scopes, ",")
 	for i := range optScopes {
 		optScopes[i] = strings.TrimSpace(optScopes[i])
 	}
 
-	env, err := envConfig(*useImplicit || *usePKCE)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%s\n\n", err)
-		return
+	var env map[string]interface{}
+	var tp *oidc.TestProvider
+	if *useTestProvider {
+		l, err := oidc.NewTestingLogger(hclog.New(nil))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n\n", err)
+			return
+		}
+		// Generate a key to sign JWTs with throughout most test cases
+		priv, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
I don't feel strongly about this, but you may want to not use RSA2048. I realize it's always available because it's required by the standard, but it's also not super desirable. So given this is an example we may want to encourage more secure options.